src/java.base/share/classes/java/lang/reflect/code/bytecode/SlotSSA.java

```diff
@@ -164,14 +160,13 @@ static void variableToValue(Node n,
                                 Map<Block.Reference, List<SlotValue>> joinSuccessorValues) {
 
         int size = n.b().ops().size();
-
         // Check if slot is associated with block argument (phi)
         // Push argument onto slot's stack
         {
             Set<Integer> slots = joinPoints.get(n.b());
             if (slots != null) {
                 slots.forEach(slot -> {
-                    variableStack.get(slot).push(new SlotBlockArgument(n.b(), slot));
+                    variableStack.computeIfAbsent(slot, _ -> new ArrayDeque<>()).push(new SlotBlockArgument(n.b(), slot));
```

Comment:
Regular SSA variable must be declared in the dominant block, so it is always present on the variableStack.
However SlotSSA slot is stored without any dominant declaration, so my assumption is that it may be missing on the  SlotSSA variable stack.
This difference caused race conditional NPEs (randomly dependent on the dominance tree node children order).

Reply:
Suggest you add a comment in the source here and with line 192 (for the var SSA code i placed in asserts because the var declaration dominates, but comments will have to do here)